lib/aws.js

```diff
@@ -33,8 +33,7 @@ AWS.util.update(AWS.Config.prototype.keys, {
     var credentials = null;
     new AWS.CredentialProviderChain([
       function () { return new AWS.EnvironmentCredentials('AWS'); },
-      function () { return new AWS.EnvironmentCredentials('AMAZON'); },
-      function () { return new AWS.SharedIniFileCredentials(); }
```

Comment:
@chrisradek Is there anything you want me to change here? The `credentials` method doesn't work for assume role profiles (it will always return null in that case). A couple of things I can think of, if just removing the provider is unacceptable:

1. Add a configuration option to the provider to disable assume role. It doesn't work in this case anyway. This seems bad to me, but would result in the same behavior as before my change.
2. Separate the normal roles and the assume roles into different providers. I don't think this is a good idea since it requires parsing the creds file twice and separates the parsing code. Botocore does this, but only because the code was copy+pasted from the cli.
3. Somehow make this method complete asynchronously. I don't know if this is feasible or not. I haven't looked into it.

For options 2 & 3, there is the risk that you get unnecessary assume role API requests if the user doesn't use the default credentials configuration. Those unnecessary API requests could slow down SDK initialization a bit.